src/zenml/models/component_models.py

```diff
@@ -65,6 +80,76 @@ class ComponentResponseModel(ComponentBaseModel, ShareableResponseModel):
     ANALYTICS_FIELDS: ClassVar[List[str]] = ["type", "flavor"]
 
 
+# ------ #
+# FILTER #
+# ------ #
+
+
+class ComponentFilterModel(ShareableProjectScopedFilterModel):
+    """Model to enable advanced filtering of all ComponentModels.
+
+    The Component Model needs additional scoping. As such the `_scope_user`
+    field can be set to the user that is doing the filtering. The
+    `generate_filter()` method of the baseclass is overwritten to include the
+    scoping.
+    """
+
+    FILTER_EXCLUDE_FIELDS: ClassVar[List[str]] = [
+        *ShareableProjectScopedFilterModel.FILTER_EXCLUDE_FIELDS,
+        "_scope_type",
+    ]
+    CLI_EXCLUDE_FIELDS: ClassVar[List[str]] = [
+        *ShareableProjectScopedFilterModel.CLI_EXCLUDE_FIELDS,
+        "_scope_type",
+    ]
+    _scope_type: str = PrivateAttr(None)
+
+    is_shared: Union[bool, str] = Query(
+        default=None, description="If the stack is shared or private"
+    )
+    name: str = Query(
+        default=None,
+        description="Name of the stack component",
+    )
+    flavor: str = Query(
+        default=None,
+        description="Flavor of the stack component",
+    )
+    type: str = Query(
+        default=None,
+        description="Type of the stack component",
+    )
+    project_id: Union[UUID, str] = Query(
+        default=None, description="Project of the stack"
+    )
+    user_id: Union[UUID, str] = Query(None, description="User of the stack")
```

Comment:
@AlexejPenner Why does this have both `_scope_type` and `type`? Similarly, I noticed many project-scoped filter models like this one also have both `project_id`, `user_id` and `_scope_project`, `_scope_user`. I didn't fully understand yet when you would use which here.

Reply:
Its two different things the scope is always a separate circle in the Venn diagram of conditions (so Scope ∩ Filter is what you get) - However if you want to filter with an or operator within the filter you can always do this with type (e.g Filter=(type  ∪ flavor ∪ is_shared).

Reply:
Ok makes sense, I think. So the point is that project/user scoped resources usually always require an AND filter on those entities and should still allows OR filters on other fields, but perhaps in some other queries you don't have this scoping and want to also include project/user in an OR filter?